devices/Rtc/Devices/Pcf85263/Pcf85263.cs

```diff
@@ -0,0 +1,455 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Pcf85263"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        public Pcf85263(I2cDevice i2cDevice) // TODO: ensure default parameters!
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+        }
+
+        /// <summary>
+        /// Read time from the Pcf85263.
+        /// </summary>
+        /// <returns>Time from the device.</returns>
+        protected override DateTime ReadTime()
+        {
+            // Sec, Min, Hour, Date, Day, Month & Century, Year
+            SpanByte readBuffer = new byte[7];
+
+            _i2cDevice.WriteByte((byte)Rtc.PCF85263RtcRegisters.RTC_SECOND_ADDR);
+            _i2cDevice.Read(readBuffer);
+
+            return new DateTime(
+                ((1900 + (readBuffer[5] >> 7)) * 100) + NumberHelper.Bcd2Dec(readBuffer[6]),
+                NumberHelper.Bcd2Dec((byte)(readBuffer[5] & 0b0001_1111)),
+                NumberHelper.Bcd2Dec((byte)(readBuffer[3] & 0b0011_1111)),
+                NumberHelper.Bcd2Dec((byte)(readBuffer[2] & 0b0011_1111)),
+                NumberHelper.Bcd2Dec((byte)(readBuffer[1] & 0b0111_1111)),
+                NumberHelper.Bcd2Dec((byte)(readBuffer[0] & 0b0111_1111)));
+        }
+
+        /// <summary>
+        /// Set Pcf85263 Time.
+        /// </summary>
+        /// <param name="time">Time.</param>
+        protected override void SetTime(DateTime time)
+        {
+            SpanByte writeBuffer = new byte[8];
+
+            writeBuffer[0] = (byte)Rtc.PCF85263RtcRegisters.RTC_SECOND_ADDR;
+
+            // Set bit8 as 0 to guarantee clock integrity
+            writeBuffer[1] = (byte)(NumberHelper.Dec2Bcd(time.Second) & 0b0111_1111);
+            writeBuffer[2] = NumberHelper.Dec2Bcd(time.Minute);
+            writeBuffer[3] = NumberHelper.Dec2Bcd(time.Hour);
+            writeBuffer[4] = NumberHelper.Dec2Bcd(time.Day);
+            writeBuffer[5] = NumberHelper.Dec2Bcd((int)time.DayOfWeek);
+            if (time.Year >= 2000)
+            {
+                writeBuffer[6] = (byte)(NumberHelper.Dec2Bcd(time.Month) | 0b1000_0000);
+                writeBuffer[7] = NumberHelper.Dec2Bcd(time.Year - 2000);
+            }
+            else
+            {
+                writeBuffer[6] = NumberHelper.Dec2Bcd(time.Month);
+                writeBuffer[7] = NumberHelper.Dec2Bcd(time.Year - 1900);
+            }
+
+            _i2cDevice.Write(writeBuffer);
+        }
+
+        /// <summary>
+        /// Cleanup.
+        /// </summary>
+        /// <param name="disposing">Is disposing.</param>
+        protected override void Dispose(bool disposing)
+        {
+            _i2cDevice?.Dispose();
+            ////_i2cDevice = null!;
+
+            base.Dispose(disposing);
+        }
+
+        ////public void DisableClockOutEnableInterruptA()
```

Comment:
all the commented code is work in progress? Otherwise please clean it!

Reply:
Thanks. Although WiP, that has stalled and it not required for the minimum functionality, so removed.